core/upgrader/impl/src/services/install_canister.rs

```diff
@@ -39,6 +37,29 @@ impl InstallCanister for StationDisasterRecoveryInstall {
         arg: Vec<u8>,
         mode: InstallMode,
     ) -> Result<(), String> {
+        // For install and reinstall, we need to make the station self controlled.
+        match mode {
+            InstallMode::Install | InstallMode::Reinstall => {
+                mgmt::update_settings(mgmt::UpdateSettingsArgument {
+                    canister_id: canister_id.clone(),
+                    settings: mgmt::CanisterSettings {
+                        controllers: Some(vec![canister_id, ic_cdk::id()]),
```

Comment:
This would wipe the fallback controller if there's one.

Reply:
The station canister would do that anyway in the timer that the init hook creates [here](https://github.com/dfinity/orbit/blob/9497822f35d9361decff83d2afb34af97cc77e57/core/station/impl/src/services/system.rs#L171)

Reply:
I see. So it isn't necessarily a bug after all, but if the call you linked fails, then the fallback controller is wiped for good, so it might make sense to preserve it.

Reply:
That makes sense, i'll add it here and will also extend the test mentioned below

Reply:
The upgrader can either remember the fallback controller in its state or query it using canister_status.